src/Symfony/Component/Serializer/Tests/Normalizer/NumberNormalizerTest.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Tests\Normalizer;
+
+use BcMath\Number;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Serializer\Exception\InvalidArgumentException;
+use Symfony\Component\Serializer\Exception\NotNormalizableValueException;
+use Symfony\Component\Serializer\Normalizer\NumberNormalizer;
+
+/**
+ * @requires PHP 8.4
+ * @requires extension bcmath
+ */
+class NumberNormalizerTest extends TestCase
+{
+    private NumberNormalizer $normalizer;
+
+    protected function setUp(): void
+    {
+        $this->normalizer = new NumberNormalizer();
+    }
+
+    public function testSupportsNormalization()
+    {
+        $this->assertTrue($this->normalizer->supportsNormalization(new Number('1.23')));
+        $this->assertFalse($this->normalizer->supportsNormalization((object) ['value' => '1.23', 'scale' => 2]));
+        $this->assertFalse($this->normalizer->supportsNormalization('1.23'));
+        $this->assertFalse($this->normalizer->supportsNormalization(1.23));
+        $this->assertFalse($this->normalizer->supportsNormalization(null));
+    }
+
+    public function testNormalize()
+    {
+        $this->assertSame('1.23', $this->normalizer->normalize(new Number('1.23')));
+        $this->assertSame('1', $this->normalizer->normalize(new Number('1')));
+        $this->assertSame('123', $this->normalizer->normalize(new Number(123)));
+    }
+
+    public function testNormalizeBadObjectTypeThrowsException()
+    {
+        $this->expectException(InvalidArgumentException::class);
+        $this->normalizer->normalize(new \stdClass());
+    }
+
+    public function testSupportsDenormalization()
+    {
+        $this->assertTrue($this->normalizer->supportsDenormalization(null, Number::class));
+        $this->assertFalse($this->normalizer->supportsDenormalization(null, \stdClass::class));
+    }
+
+    public function testDenormalize()
+    {
+        $this->assertEquals(new Number('1.23'), $number = $this->normalizer->denormalize('1.23', Number::class));
+        $this->assertEquals(new Number('123'), $this->normalizer->denormalize('123', Number::class));
+        $this->assertEquals(new Number('123'), $this->normalizer->denormalize(123, Number::class));
+    }
```

Comment:
Same here, these are three tests. Also, what happens if I denormalize `null`? The previous test says it's supported.

Reply:
Thanks, added `null` check to `supportsDenormalization`.